### PR TITLE
Set up collecting crash dumps for desktop tests

### LINKED ISSUE #5060: Set up desktop tests to capture and upload crash dumps
e2etest does this but the desktop tests don't, and they're crashing intermittently (#5059). This tracks capturing desktop test crash dumps.

### PR DESCRIPTION
**Pull request by asklar**
Fixes #5060 
Collect crash dumps for desktop unit tests

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5069)

### CODE REVIEW

File: .ado/windows-vs-pr.yml [hunk 1]
```diff
@@ -455,7 +487,8 @@ jobs:
           platform: $(BuildPlatform)
           configuration: $(BuildConfiguration)
           publishRunAttachments: true
-          collectDumpOn: onAbortOnly
+          collectDumpOn: always
```
Comment (vmoroz) [inline .ado/windows-vs-pr.yml:490]: >always [](start = 25, length = 6)

I wonder what it means to collect dumps always. Does it collect on successful runs too?

File: .ado/windows-vs-pr.yml [hunk 2]
```diff
@@ -423,10 +453,11 @@ jobs:
...
+          diagnosticsEnabled: true
           vsTestVersion: latest
           otherConsoleOptions: '/blame -- RunConfiguration.TestSessionTimeout=300000'
-        condition: and(succeeded(), ne(variables['BuildConfiguration'], 'Debug'))
+#        condition: and(succeeded(), ne(variables['BuildConfiguration'], 'Debug'))
```
Comment (dannyvv) [inline .ado/windows-vs-pr.yml:460]: ># [](start = 0, length = 1)

intentional?

File: .ado/windows-vs-pr.yml (hunk 1 above)
Comment (dannyvv) [reply .ado/windows-vs-pr.yml:490]: From the [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/vstest?view=azure-devops)


collectDumpOnCollect process dump and attach to test run report | (Optional) Use this option to collect a mini-dump that can be used for further analysis.On abort only: mini-dump will be collected only when test run is aborted.Always: mini-dump will always be collected regardless of whether the test run completes or not.Never: mini-dump will not be collected regardless of whether the test run completes or not
-- | --

**Pull request closed without merging**